django/db/backends/oracle/operations.py

```diff
@@ -512,29 +491,44 @@ def combine_expression(self, connector, sub_expressions):
             return 'POWER(%s)' % ','.join(sub_expressions)
         return super().combine_expression(connector, sub_expressions)
 
-    def _get_sequence_name(self, table):
+    def _get_no_autofield_sequence_name(self, table):
+        """
+        Manually created sequence name to keep backward compatibility for
+        AutoFields that aren't Oracle identity columns.
+        """
         name_length = self.max_name_length() - 3
         sequence_name = '%s_SQ' % strip_quotes(table)
         return truncate_name(sequence_name, name_length).upper()
 
-    def _get_trigger_name(self, table):
-        name_length = self.max_name_length() - 3
-        trigger_name = '%s_TR' % strip_quotes(table)
-        return truncate_name(trigger_name, name_length).upper()
+    def _get_sequence_name(self, cursor, table, pk_name):
+        cursor.execute("""
+            SELECT sequence_name
+            FROM user_tab_identity_cols
+            WHERE table_name = UPPER(%s)
+            AND column_name = UPPER(%s)""", [table, pk_name])
+        row = cursor.fetchone()
+        if row is None:
+            return self._get_no_autofield_sequence_name(table)
+        else:
+            return row[0]
 
     def bulk_insert_sql(self, fields, placeholder_rows):
         query = []
         for row in placeholder_rows:
             select = []
             for i, placeholder in enumerate(row):
                 # A model without any fields has fields=[None].
-                if not fields[i]:
-                    select.append(placeholder)
-                else:
+                if fields[i]:
                     internal_type = getattr(fields[i], 'target_field', fields[i]).get_internal_type()
-                    select.append(BulkInsertMapper.types.get(internal_type, '%s') % placeholder)
+                    placeholder = BulkInsertMapper.types.get(internal_type, '%s') % placeholder
+                # Add columns aliases to the first select to avoid "ORA-00918:
+                # column ambiguously defined" when two or more columns in the
+                # first select has the same values.
+                if not query:
+                    placeholder = "%s col_%s" % (placeholder, i)
+                select.append(placeholder)
             query.append('SELECT %s FROM DUAL' % ', '.join(select))
-        return ' UNION ALL '.join(query)
+        return 'SELECT * FROM (%s)' % ' UNION ALL '.join(query)
```

Comment:
If we use Oracle identity columns in tables in which we try to bulk insert then Oracle internally adds `sequence.nextval` to it. On the other hand we cannot use `sequence.nextval` in the following construct: _"A SELECT statement that is combined with another SELECT statement with the UNION, INTERSECT, or MINUS set operator"_ (see [documentation](http://docs.oracle.com/database/121/SQLRF/pseudocolumns002.htm#SQLRF00253)), hence I changed statement e.g.:
```sql
SELECT 'foo', 'foo', 'foo_sth' FROM dual
UNION ALL
SELECT 'bar', 'bar', 'bar_sth' FROM dual
```
to 
```sql
SELECT * FROM
  (SELECT 'foo' col_0, 'foo' col_1, 'foo_sth' col_2 FROM dual
   UNION ALL
   SELECT 'bar', 'bar', 'bar_sth' FROM dual)
```
I also added columns aliases to the first select to avoid `ORA-00918: column ambiguously defined` when two or more columns in the first select has the same values.